vasyl_varyshchuk/Task_2/pass_arg_decorator.py

```diff
@@ -0,0 +1,35 @@
+"""
+This module implements a retry decorator with ability to pass a parameter into decorator
+"""
+
+
+def decorator(retries=None):
+
+    def my_decorator(func):
+
+        def func_wrapper():
+            if retries is not None:
+                i = 0
+                while i < retries:
```

Comment:
It looks like this function will work in this way:
if retries = 4
- first run -  failed
 - the second run - done

It will also run 3 and 4 times what is not expected.

Reply:
Thanks, I have understood my mistake. Appropriate corrections were made.